app/services/auth_service.rb

```diff
@@ -0,0 +1,45 @@
+class AuthService
+  attr_reader :auth, :signed_in_resource
+  attr_reader :user
+  TEMP_EMAIL_PREFIX = 'change@me'.freeze
+
+  def initialize(auth, signed_in_resource = nil, user = nil)
+    @auth = auth
+    @signed_in_resource = signed_in_resource
+    @user = user
+  end
+
+  def find_for_oauth
+    identity = Identity.find_for_oauth(@auth)
+    @user = user_select(identity)
+    @user = user_create unless @user.present?
+    identity.add_identity(@user)
+    @user
+  end
+
+  private
+
+  def user_select(identity)
+    @signed_in_resource ? @signed_in_resource : identity.user
+  end
+
+  def user_create
+    # user = User.where(email: email).first if email
+    # return if User.where(email: email).first.present?
+    user = User.new fill_user_fields
+    UsersHelper.user_save(user)
```

Comment:
you can include this helper in service if you need it

Reply:
[31]:UtilityFunction: AuthService#user_save doesn't depend on instance state (maybe move it to another class?) [https://github.com/troessner/reek/blob/v5.0.2/docs/Utility-Function.md]